guessing-game-v2/contracts/GuessingGame.sol

```diff
@@ -22,51 +22,50 @@ contract GuessingGame {
 
     constructor() {
         owner = msg.sender;
-        _resetSecretNumber();
+        _setSecretNumber();
     }
 
-    function guess(uint256 _number) external payable {
-        require(_number > 0 && _number <= MAX_GUESS, 'Secret number should be between 1 and 1000');
-        require(msg.value == GUESS_FEE, 'You need to pay a fees of 0.443 ETH to make a guess');
-        totalGuesses += 1;
+function guess(uint256 _number) external payable {
+    require(_number > 0 && _number <= MAX_GUESS, 'Secret number should be between 1 and 1000');
+    require(msg.value == GUESS_FEE, 'You need to pay a fees of 0.443 ETH to make a guess');
+    totalGuesses += 1;
 
-        if (lastGuess[msg.sender] == 0 || lastResetTime > lastGuess[msg.sender]) {
-            lastGuess[msg.sender] = 0;
-            lastDifference[msg.sender] = 0;
-        }
+    if (lastGuess[msg.sender] == 0 || lastResetTime > lastGuess[msg.sender]) {
+        lastDifference[msg.sender] = MAX_GUESS + 1;
+    }
 
-        uint256 currentDifference = _number > secretNumber ? _number - secretNumber : secretNumber - _number;
-        string memory feedback;
+    uint256 currentDifference = _number > secretNumber ? _number - secretNumber : secretNumber - _number;
```

Comment:
`currentDifference` should be inside the `} else {` because we dont need to calculate it here.